active-rfcs/0000-scoped-styles-changes.md

```diff
@@ -0,0 +1,116 @@
+- Reference Issues: N/A
+- Implementation PR: N/A
+
+# Summary
+
+Provide more consistent custom CSS extensions in Single File Component scoped styles.
+
+# Basic example
+
+``` html
+<style scoped>
+/* deep selectors */
+::v-deep(.foo) {}
+
+/* targeting slot content */
+::v-slotted(.foo) {}
+
+/* one-off global rule */
+::v-global(.foo) {}
+</style>
+```
+
+# Motivation
+
+Vue SFC scoped styles makes the CSS apply to the current component only. There are a number of cases that users commonly run into that can be improved.
+
+## Deep Selectors
+
+Sometimes we may want to explicitly make a rule target child components.
+
+Originally we supported the `>>>` combinator to make the selector "deep". However, some CSS pre-processors such as SASS has issues parsing it since this is not an official CSS combinator.
+
+We later switched to `/deep/`, which was once an actual proposed addition to CSS (and even natively shipped in Chrome), but later dropped. This caused confusion for some users, since they worry that using `/deep/` in Vue SFCs would make their code not supported in browsers that have dropped the feature. However, just like `>>>`, `/deep/` is only used as a compile-time hint by Vue's SFC compiler to rewrite the selector, and is removed in the final CSS.
+
+To avoid the confusion of the dropped `/deep/` combinator, we introduced yet another custom combinator, `::v-deep`, this time being more explicit that this is a Vue-specific extension, and using the pseudo-class syntax so that any pre-processor should be able to parse it.
+
+The previous versions of the deep combinator are still supported for compatibility reasons in the current [Vue 2 SFC compiler](https://github.com/vuejs/component-compiler-utils), which again, can be confusing to users. In v3, we are deprecating the support for `>>>` and `/deep/`.
+
+As we were working on the new SFC compiler for v3, we noticed that CSS pseudo classes are in fact semantically NOT [combinators](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Selectors/Combinators). It is more consistent with idiomatic CSS for pseudo classes to accept arguments instead, so we are also making `::v-deep()` work that way. The current usage of `::v-deep` as a combinator is still supported, however it is considered deprecated and will raise a warning.
+
+## Targeting / Avoiding Slot Content
+
+Currently, slot content passed in from the parent are affected by both the parent's scoped styles AND the child's scoped styles. There is no way to author rules that explicitly target slot content only, or ones that do not affect slot content.
+
+In v3, we intend to make child scoped styles NOT affecting slot content by default. To explicitly target slot content, the `::v-slotted()` pseudo class can be used.
+
+## One-Off Global Rules
+
+Currently to add a global CSS rule we need to use a separate unscoped `<style>` block. We are introducing a new `::v-global()` pseudo class for one-off global rules.
```

Comment:
does this bring any improvement in performance or anything? I find the separate `style` easier to read and more intuitive because it's just like a regular `style` tag.
Unless `scoped` becomes the default but that would be even harder to migrate

Reply:
Technically, yes. For webpack, an extra `<style>` block becomes a separate module with a whole pipeline (i.e. all the loaders configured for CSS are invoked for this block again). Putting them in the same block improves bundler build performance.

Reply:
That makes sense, I think it's worth indicating that as a benefit